src/key_generator/util.rs

```diff
@@ -0,0 +1,135 @@
+//! # Helper methods for Secret key generator
```

Comment:
Util not a good name, really, normally it's a symptom of bad architecture, try to find better decomposition scheme.